### PR TITLE
Add some missing tests for NuGetAudit

### PR DESCRIPTION
**Pull request by zivkan**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/2253

Regression? No

## Description

Adds some missing tests for NuGetAudit functionality.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [x] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

File: test/NuGet.Core.Tests/NuGet.ProjectModel.Test/RestoreAuditPropertiesTests.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    public class RestoreAuditPropertiesTests
+    {
+        [Fact]
+        public void Equals_WithNull_ReturnsFalse()
+        {
+            // Arrange
+            var properties = new RestoreAuditProperties();
+
+            // Act & Assert
+            Assert.False(properties.Equals((RestoreAuditProperties?)null));
```
Comment (jeffkl) [inline test/NuGet.Core.Tests/NuGet.ProjectModel.Test/RestoreAuditPropertiesTests.cs]: Should all new tests use FluentAssertions?

File: test/NuGet.Core.Tests/NuGet.ProjectModel.Test/RestoreAuditPropertiesTests.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            Assert.False(properties.Equals((object?)null));
+            Assert.False(properties == null);
+            Assert.False(null == properties);
+            Assert.True(properties != null);
+            Assert.True(null != properties);
+        }
+
+        [Fact]
+        public void Euqals_WithSameInstance_ReturnsTrue()
```
Comment (jeffkl) [inline test/NuGet.Core.Tests/NuGet.ProjectModel.Test/RestoreAuditPropertiesTests.cs]:
```suggestion
        public void Equals_WithSameInstance_ReturnsTrue()
```

Comment (zivkan) [reply test/NuGet.Core.Tests/NuGet.ProjectModel.Test/RestoreAuditPropertiesTests.cs]: I swear I was staring at this for over an hour, over the course of a few days, until I finally saw what the actual difference was 👓

**Review by jeffkl — APPROVED** [review]
💯

**Pull request merged**